toolbox/src/DiscreteFilter.cpp

```diff
@@ -0,0 +1,317 @@
+#include "DiscreteFilter.h"
+#include "BlockInformation.h"
+#include "Error.h"
+#include "ForwardKinematics.h"
+#include "Signal.h"
+#include <algorithm>
+#include <cassert>
+#include <iCub/ctrl/filters.h>
+#include <sstream>
+#include <string>
+#include <yarp/sig/Vector.h>
+
+// Parameters
+#define PARAM_IDX_FLT_TYPE 1 // ::Filter type
+#define PARAM_IDX_NUMCOEFF 2 // ::Filter numerator coefficients
+#define PARAM_IDX_DENCOEFF 3 // ::Filter denominator coefficients
+#define PARAM_IDX_1LOWP_FC 4 // ::FirstOrderLowPassFilter cut frequency
+#define PARAM_IDX_1LOWP_TS 5 // ::FirstOrderLowPassFilter sampling time
+#define PARAM_IDX_MD_ORDER 6 // ::MedianFilter order
+#define PARAM_IDX_INIT_Y0 7  // Output initialization
+#define PARAM_IDX_INIT_U0 8  // ::Filter input initialization
+
+// Inputs
+#define INPUT_IDX_SIGNAL 0
+// Outputs
+#define OUTPUT_IDX_SIGNAL 0
+// Other defines
+#define SIGNAL_DYNAMIC_SIZE -1
+
+using namespace wbt;
+using namespace iCub::ctrl;
+using namespace yarp::sig;
+
+std::string DiscreteFilter::ClassName = "DiscreteFilter";
+
+DiscreteFilter::DiscreteFilter() : filter(nullptr), inputSignalVector(nullptr)
+{
+    y0 = new Vector(0);
```

Comment:
As the delete are done in the terminate, move the allocation in the corresponding method, and not in the constructor.
Here simply initialise to `nullptr` the pointers